### PR TITLE
Update xgboost titanic tutorial to xgboost==0.81

### LINKED ISSUE #289: Obsolete documentation - tutorial
Hi, 
code in tutorial does not work with latest XGBoost 0.8.1.

Location of the problem:

1. Go here:  https://eli5.readthedocs.io/en/latest/tutorials/xgboost-titanic.html
2. Go to Section 3: Explaining weights - there is code below

```
booster = clf.booster()  # <----- method booster() does not exist anymore
original_feature_names = booster.feature_names
booster.feature_names = vec.get_feature_names()
print(booster.get_dump()[0])
# recover original feature names
booster.feature_names = original_feature_names
```

It throws exception:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-45-2800cf4b424a> in <module>
----> 1 booster = clf.booster()
      2 original_feature_names = booster.feature_names
      3 booster.feature_names = vec.get_feature_names()
      4 print(booster.get_dump()[0])
      5 # recover original feature names

TypeError: 'str' object is not callable
```
which makes sense, because `clf.booster` returns string` 'gbtree'`

### PR DESCRIPTION
**Pull request by lopuhin**
Fixes #289

- use xgboost==0.81
- no warnings any more, remove their filtering
- the issue with xgboost + sparse matrices is no longer happening, so remove ``CSCTransformer``

Also I happened to have a newer pandoc version, which produces slightly different output (adjusted ``sed`` scripts accordingly) - @kmike what do you think, shall we stick to 1.x instead (nbconvert warns that is supports only 1.x)?

### CODE REVIEW

**Comment by kmike** [comment]
It seems conversion works with pandoc 2.0, and not only here - https://github.com/jupyter/nbconvert/issues/928 - it looks like there are issues, but they're not that important for us. So I think that's fine not to stick with pandoc 1.x. What do you think about documenting somewhere (maybe in a comment in a script) that the script needs pandoc 2+?

**Comment by kmike** [comment]
A great clean up, thanks @lopuhin!

**Comment by lopuhin** [comment]
Thanks for review @kmike !

> What do you think about documenting somewhere (maybe in a comment in a script) that the script needs pandoc 2+?

I added that to ``docs/README.rst``, and applied new pandoc to all notebooks in #292
